v3/onesignal-metabox/onesignal-metabox.js

```diff
@@ -32,4 +32,25 @@ window.addEventListener("DOMContentLoaded", () => {
     setDisplay(customiseWrap, customisePost.checked);
     setDisabled(customiseWrapChild, !customisePost.checked);
   });
+
+  // Watch for the Publish button to be added to the DOM
+  const observer = new MutationObserver((mutations, obs) => {
+    const publishButton = document.querySelector('.editor-post-publish-button__button');
+
+    if (publishButton) {
+      publishButton.addEventListener('click', function() {
+        setTimeout(() => {
```

Comment:
A wait is needed here, otherwise the notification won't send. This may because it takes the plugin time to structure the request and immediately unchecking it will mark it as "don't send". Generally I would prefer that functionality not be dependent on waiting. Happy to hear any thoughts or feedback on this.